### PR TITLE
[#3] [#7] [iOS] Set Up CICD for iOS for test and upload to Firebase Distribution

### LINKED ISSUE #3: [iOS] Set Up CICD for iOS for test and upload to Firebase Distribution
## Why

Set up GitHub Actions to build the iOS app and publish to Firebase Distribution for CD.

## Acceptance Criteria

- Run GitHub Actions workflow when `push`.
- Run unit test for iOS.
- Build iOS for adhoc signing.
- Submit iOS to Firebase Distribution.

### LINKED ISSUE #7: Set Up Firebase Distribution for iOS app
## Why

Use Firebase Distribution for testing the app.

## Acceptance Criteria

- Create match repo.
- Sync match repo with Nimble's certificates.
- Create 1Password credentials.
- Create Firebase project.
- Publish the iOS app to Firebase Distribution.
- No need to install Firebase library.

### PR DESCRIPTION
**Pull request by blyscuit**
close #3 
close #7 

## What Happened

Add two workflows:
- Deploy Staging iOS app to Firebase Distribution.
- Deploy Production iOS app to Firebase Distribution.

Both workflows:
- Test debug iOS app.
- Match certificates and profiles from github.com/nimblehq/match-certificates

## Insight

- Modified iOS template to use with KMM.
- Use Firebase only for fastlane.

## Proof of Work

Tested link https://github.com/blyscuit/kmm-swiftUI-IC/actions/runs/3058076997

Screen Shot 2022-09-15 at 13 09 57

### CODE REVIEW

**Review by suho — APPROVED** [review]
LGTM

File: .gitignore [hunk 1]
```diff
@@ -70,22 +70,23 @@ Pods/
...
-DerivedData
+iosApp/DerivedData
 
 # Tuist
 .tuist-generated
 
 # JetBrains IDEs
 
 .idea
+GoogleService-*.plist
```
Comment (minhnimble) [inline .gitignore]: Isn't this file is specific for iOS and we should placed it in `iosApp/`, right?

**Review by minhnimble — APPROVED** [review]
one minor concern, the rest LGTM 🚀

File: .gitignore (hunk 1 above)
Comment (blyscuit) [reply .gitignore]: Modified the gitignore file. 🙏

**Review by edgarss — APPROVED** [review]

**Pull request merged**